### PR TITLE
Update Chart.js Version

### PR DESCRIPTION
**Pull request by ivelinmegdanov**
**Updates Chart.js to the latest version. The changes include updating the Gulp tasks and modifying the code to accommodate any breaking changes introduced in the new version.**

- Updated: Chart.js to the latest version.
- Modified: Gulp tasks to copy the new version of Chart.js.
- Updated: Chart initialization code to use the new Chart.js API.

### CODE REVIEW

**Comment by superKalo** [comment]
In the migrated (new) version, the tooltips that display min and max price per selected period are missing:
 
Screenshot 2024-07-24 at 22 54 23

**Comment by superKalo** [comment]
In the migrated version, the tooltip styles look a bit messed up. How it looks now:

Screenshot 2024-07-24 at 22 58 46

vs before:

Screenshot 2024-07-24 at 22 58 53

File: src/js/chart.js [hunk 1]
```diff
@@ -5,146 +5,137 @@ window.App = window.App || {};
...
-                return;
-            }
-
+window.App.Chart.prototype.alwaysVisibleTooltipsPlugin = function () {
+    const plugin = {
+        id: 'alwaysVisibleTooltips',
+        beforeRender: (chart) => {
             if (chart.config.options.showAllTooltips) {
-                // create an array of tooltips
-                // we can't use the chart tooltip because there is only one tooltip per chart
```
Comment (superKalo) [inline src/js/chart.js:137]: Hm, why you deleted this comment?

**Comment by superKalo** [comment]
There is an annoying hover effect when you hover over the data set:

https://github.com/user-attachments/assets/afa15650-e5bb-43a7-8168-f8d9c87e95a0

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   },
   "dependencies": {
     "axios": "0.16.2",
-    "chart.js": "2.6.0",
+    "chart.js": "4.0.1",
```
Comment (superKalo) [inline package.json:13]: Why this specific version? ChartJS v4.0.1 was released 2 years ago. The latest version is v4.4.3. Does migrating to the latest version give you any trouble?

**Review by superKalo — CHANGES_REQUESTED** [review]
So far so good! I added some feedback, please take a look when you have time :v:

**Comment by superKalo** [comment]
Okay, it's a lot better now! I like the approach for displaying min and max price!

Can you please try to improve a bit the drawing of the permanent min and max tooltips?

They are just a tiny bit off in terms of 1) The triangle has a tiny space (probably less than 1px, but annoying) and 2) The text is not vertically centered (it feels just a bit more on the top).

Here's a permanent tooltip:

Screenshot 2024-08-14 at 0 44 16

vs a tooltip:

Screenshot 2024-08-14 at 0 44 32

**Comment by ivelinmegdanov** [comment]
> Okay, it's a lot better now! I like the approach for displaying min and max price!
> 
> Can you please try to improve a bit the drawing of the permanent min and max tooltips?
> 
> They are just a tiny bit off in terms of 1) The triangle has a tiny space (probably less than 1px, but annoying) and 2) The text is not vertically centered (it feels just a bit more on the top).
> 
> Here's a permanent tooltip:
> Screenshot 2024-08-14 at 0 44 16
> 
> vs a tooltip from the lib (displays perfectly):
> Screenshot 2024-08-14 at 0 44 32

Regarding the text, it appears to me that it’s consistent between both the permanent and hover tooltips,

Hover tooltip: 
image

Permanent tooltip:
image

Overlap on both:
image
image

**Review by bego1996 — APPROVED** [review]

**Review by superKalo — APPROVED** [review]
All right, now it looks perfect 💎 Thank you!

**Pull request merged**